### DISCUSSION TITLE
Tinybird MCP Server: large-scale real-time analytics in Claude

### DISCUSSION THREAD

**Discussion by sdairs** [modelcontextprotocol/.github#42] [Show and tell]
### Pre-submission Checklist

- [X] This post follows the MCP community guidelines

### What would you like to share?

We've built an [MCP server](https://github.com/tinybirdco/tinybird_mcp_claude) for [Tinybird](https://www.tinybird.co/)

You can connect it to any Tinybird Workspace, it will auto-discover existing tables/queries, and then help guide you through exploring your data. Tinybird is built for analytics at very high scale (many billions of rows, thousands of concurrent queries, and millisecond latencies) so its pretty interesting to have that kind of analytical capability inside a chat experience, as its not something that can easily be done locally on the users machine. 

In the analytics space, we've seen a lot of AI being used to generate SQL for known questions, which is pretty handy but we think its only one part of the puzzle. One of the challenges we often see with data, particularly at high volume, is that it can become hard to understand and explore, and there's a lot of interesting insights that people just haven't thought of...So that's what we've started to explore with our `v0` server. Providing a guided experience where the AI understands not just the database syntax, but has your specific data context, and can make suggestions about what to explore, other dimensions around what you're exploring, and quickly summarise what it found. The idea is not so much to create reports, but to give the users ideas about valuable things that are in their data that they might want to build with.

It's borrows heavily from the SQLite example, so thank you for creating that!

### Relevant Links

- https://www.tinybird.co/blog-posts/claude-analyze-bluesky-data-tinybird-mcp-server
- https://github.com/tinybirdco/tinybird_mcp_claude/tree/main

**Comment by dsp-ant** [comment]
Very cool. Thanks for building this!
